src/transformers/hygraph.ts

```diff
@@ -0,0 +1,121 @@
+import {
+	UrlGenerator,
+	UrlGeneratorOptions,
+	UrlParser,
+	UrlTransformer,
+} from "../types.ts";
+
+const hygraphRegex =
+/https:\/\/(?<region>[a-z0-9-]+)\.graphassets\.com\/(?<envId>[a-z0-9]+)(?:\/(?<transformations>.*?))?\/(?<handle>[a-z0-9]+)$/;
+
+export interface HygraphParams {
+  region?: string;
+  envId?: string;
+  transformations: Record<string, Record<string, string | number>>;
+  handle?: string;
+}
+
+export const parse: UrlParser<HygraphParams> = (
+	url,
+) => {
+  const base = url.toString();
+  const matches = base.match(hygraphRegex);
+
+  if (!matches?.length) {
+		throw new Error("Invalid Hygraph URL");
+	}
+
+  const group = matches.groups || {};
+  const { transformations: unparsedTransformations, ...params } = group;
+  const transformations = parseTransformations(unparsedTransformations || "");
+
+	return {
+		base,
+		width: Number(transformations.resize?.width) || undefined,
+		height: Number(transformations.resize?.height) || undefined,
+    format: transformations.auto_image ? "auto" : transformations.output?.format?.toString() || undefined,
+    params: { transformations, ...params },
+    cdn: "hygraph",
+	};
+};
+
+export const generate: UrlGenerator<HygraphParams> = (
+	{ base, width, height, format, params },
+) => {
+	const parsed = parse(base.toString());
+	const props: HygraphParams = {
+		transformations: {},
+		...parsed.params,
+		...params,
+	};
+
+  props.transformations.resize ||= {};
+  props.transformations.resize.fit ||= "crop";
```

Comment:
This needs to only be set if both width and height are provided. I think your previous version had a check for this. Currently this generates invalid URLs if only one dimension is provided. e.g. [this one](https://us-west-2.graphassets.com/cm2apl1zp07l506n66dmd9xo8/resize=fit:crop,width:100/auto_image/cm2tr64fx7gvu07n85chjmuno). You can catch errors like this by running `deno test --allow-net e2e.test.ts`